packages/lending/src/protocol.portfolio.ts

```diff
@@ -0,0 +1,414 @@
+  }
+
+  // more 1 BPS to ensure that transaction sending won't be hindered by discrepancies in contract calculations
+  get isHealthy() {
+    return new BigNumberJS(this.healthRate).gt(1.0001);
+  }
+
+  get netAPY() {
+    return calcNetAPY(this.totalCollateralUSD, this.positiveProportion, this.totalBorrowUSD, this.negativeProportion);
+  }
+
+  get formattedNetAPY() {
+    return formatPercentage(this.netAPY);
+  }
+
+  get formattedTotalSupplyUSD() {
+    return abbreviateUSD(this.totalSupplyUSD.toFixed());
+  }
+
+  get formattedTotalBorrowUSD() {
+    return abbreviateUSD(this.totalBorrowUSD.toFixed());
+  }
+
+  get hasPositions() {
+    return this.nonZeroSupplies.length > 0 || this.nonZeroBorrows.length > 0;
+  }
+
+  get canWithdraw() {
+    return this.totalSupplyUSD.gt(0) && this.isHealthy;
+  }
+
+  get canBorrow() {
+    return this.totalBorrowCapacityUSD.gt(0) && this.hasBorrowCapacity;
+  }
+
+  get canRepay() {
+    return this.nonZeroBorrowsForRepay.length > 0;
+  }
+
+  clone() {
+    return new Portfolio(this.chainId, this.protocolId, this.marketId, this.supplies, this.borrows);
+  }
+
+  findSupply(token: common.Token) {
+    return this.supplyMap[token.unwrapped.address] ?? this.supplyMap[token.address];
+  }
+
+  findBorrow(token: common.Token) {
+    return this.borrowMap[token.unwrapped.address] ?? this.borrowMap[token.address];
+  }
+
+  calcLeverageTimes(leverageToken: common.Token, afterPortfolio: Portfolio) {
+    let leverageTimes = '0';
+
+    if (!this.availableBorrowCapacityUSD.isZero()) {
+      const supply = this.findSupply(leverageToken);
+      const afterSupply = afterPortfolio.findSupply(leverageToken);
+      if (supply && afterSupply) {
+        const leverageUSD = new BigNumberJS(afterSupply.balance).minus(supply.balance).times(supply.price);
+        leverageTimes = leverageUSD.div(this.availableBorrowCapacityUSD).toFixed();
+      }
+    }
+
+    return leverageTimes;
+  }
+
+  calcMaxLeverageTimes(token: common.Token) {
+    let maxLeverageTimes = '0';
+
+    const supply = this.findSupply(token);
+    if (supply) {
+      maxLeverageTimes = new BigNumberJS(1).div(new BigNumberJS(1).minus(supply.ltv)).toFixed();
+    }
+
+    return maxLeverageTimes;
+  }
+
+  supply(supply: Supply): void;
+  supply(token: common.Token, amount: string): void;
+  supply(arg0: any, arg1?: any) {
+    let supply: Supply;
+    let amount: string;
+    if (isSupply(arg0)) {
+      supply = arg0;
+      amount = supply.balance;
+    } else {
+      const found = this.findSupply(arg0);
+      if (!found) return;
+      supply = found;
+      amount = arg1;
+      supply.add(amount);
+    }
+    if (Number(amount) === 0) return;
+
+    const { price, apy, usageAsCollateralEnabled, ltv, liquidationThreshold } = supply;
+
+    const supplyUSD = new BigNumberJS(amount).times(price);
+    this.totalSupplyUSD = this.totalSupplyUSD.plus(supplyUSD);
+    this.totalCollateralUSD = this.totalCollateralUSD.plus(supplyUSD);
```

Comment:
Just discussed verbally, this line should be placed inside the `if (usageAsCollateralEnabled) {` block to ensure that only collateral can be summed, and should correspondingly modify the `calcNetAPY()` function.

c.c. @Jeff-CCH

Reply:
OK

Reply:
Fixed